tests/test_norm_ops.py

```diff
@@ -136,6 +136,39 @@ def test_accuracy_layernorm(shape, dtype):
     gems_assert_close(res_bias_grad, ref_bias_grad, dtype, reduce_dim=M)
 
 
+@pytest.mark.weight_norm_interface
+@pytest.mark.parametrize("shape", REDUCTION_SHAPES)
+@pytest.mark.parametrize("dtype", FLOAT_DTYPES)
+@pytest.mark.parametrize("dim", [0, -1])
+def test_accuracy_weightnorm(shape, dtype, dim):
+    dim = dim % len(shape)
+    v = torch.randn(shape, dtype=dtype, device="cuda", requires_grad=True)
+    g = torch.randn(shape[dim], dtype=dtype, device="cuda", requires_grad=True)
+
+    ref_v = to_reference(v, True)
+    ref_g = to_reference(g, True)
```

Comment:
It seems not necessary upcast dtype here, `to_reference(False)` is enough.

Reply:
Actually, I have tested both scenarios, and there is no difference. It is set to true here to keep consistent with the LayerNorm test above. If you prefer to set it to false, that's ok.